### ISSUE TITLE
Introduce configuration file

### ISSUE THREAD

**Issue by StefMa** [ioki-mobility/summaraizer#47]
Right now it is super annoying to always put the tokens, AI model names, custom prompt etc (the flags) to each and every source and provider.
We want to support a configuration file.
The following (generic) format is possible:
```json
{
  "github": {
    "token": "abc",
    "issue": "owner/repo/1"
  },
  "google": {
    "token": "abc",
    "model": "someModel"
  },
  "openai": {
    "token": "env:OPENAI_API_KEY",
    "model": "gpt-4"
  }
}
```

It seems that using with `Viper` this is fairly easy to implement.
We can also support not only json but also different configuration languages.

We also want to support `env:ABC` as keys and read from env variables in that case.
This is a security feature for tokens that we don't want to add to a plain text file.

The configuration file can be placed in various places:
1. `./config.json`                    // Current directory
2. `$HOME/.config/appname/config.json` // XDG Base Directory
3. `$HOME/.appname/config.json`       // Legacy/simple approach

(or what Viper gives us out of the box 🤷 )

Basically the configuration can contain each and every `command` including each and every `flag` as configuration.

The config file should be "dynamically".
In case something is NOT defined, we go with the "default behaviour" on cmd (Cobra) level.

The following should give presence from top to bottom)
* CLI flag
* Config value
* CLI default
* CLI throws
